rails-snow-queen/app/assets/javascripts/quotingPanel.js

```diff
@@ -0,0 +1,49 @@
+class QuotingPanel {
+  constructor() {
+    this.polygons = {};
+    this.totalAreaInSqFt = null;
+    this.totalDue = null;
+  }
+
+  showAddress(geocodedAdress) {
+    if (geocodedAdress) {
+      let addressNode = document.getElementById("displayAddress");
+      while (addressNode.firstChild) { addressNode.removeChild(addressNode.firstChild) }
+      let splitAddress = geocodedAdress.split(",");
+      splitAddress.forEach(element => {
+        let div = document.createElement("div");
+        div.innerText = element;
+        addressNode.appendChild(div);
+      });
+      addressNode.classList.remove("hidden");
+    }
+  }
+
+  updateTotals(polygons) {
```

Comment:
This method is doing too much. Break it up into `aggregateArea`, `convertToSqFt` and  `calculatePrice`, and rename it
```javascript
handlePolygonChanged(polygons) {
  this.polygons = polygons;
  let totalAreaInFt = this.convertToSqFt(this.aggregateArea());
  let totalPrice = this.calculatePrice(totalAreaInFt);
  this.updateArea(totalAreaInFt);
  this.updatePrice(totalPrice);
  this.showQuote();
}
```